### ISSUE TITLE
StringEnum _missing_ doesn't work

### ISSUE THREAD

**Issue by tlambert03** [napari/napari#754]
## 🐛 Bug

Quick observation that is turning out to be trickier than I first thought:  The `StringEnum` subclass is not working as intended with missing values.  

1. It does not do the intended conversion to lowercase and 
2. the _real_ error message gets hidden in the stack trace, making it less informative.

```python
from napari.layers.base._constants import Blending
Blending('ADDITIVE')
#----------------------------------------
ValueError: 'ADDITIVE' is not a valid Blending
...
During handling of the above exception, another exception occurred:
...
TypeError: _missing_() missing 1 required positional argument: 'value'
```

drop a `print()` statement inside `StringEnum._missing_` and you'll see that it doesn't seem to be getting called at all.  I think that's because `Enum._missing_` is actually a `@classmethod`.  If you change `StringEnum._missing_` to a `classmethod` it does get called, however, it then becomes hard to subclass.

```python
class StringEnum(Enum):
    @classmethod
    def _missing_(cls, value):
        # cls here refers to StringEnum and NOT the subclass
        # making it hard to know if the value is valid
```
One way or the other, if `_missing_` is to be overridden, I think we need to raise a `ValueError` as the [builtin `_mising_` method](https://github.com/python/cpython/blob/2fe4c48917c2d1b40cf063c6ed22ae2e71f4cb62/Lib/enum.py#L600-L602) does.  Simply returning `self(value.lower())` does not work.

**Comment by sofroniewn** [comment]
@kevinyamauchi do you have thoughts on this?

**Comment by kevinyamauchi** [comment]
Thanks for catching this, @tlambert03 !

If I recall correctly, the original intent was to allow users to create the `StringEnum` without having to worry about the case. I totally agree that the error handling in the current implementation is confusing/broken.

In light of this, I propose that we: (1) no longer overload `_missing_` in `StringEnum` and (2) decide if we still think that it is important to make the `StringEnum` constructor case invariant. If we do want to make constructor case invariant, we can consider other approaches (e.g., [overload `__getitem__`](https://stackoverflow.com/questions/42658609/how-to-construct-a-case-insensitive-enum)).

While it is nice to be able to accept upper and lower case strings, all of our values are lowercase (given by `StringEnum._generate_next_value()`) and the strings in our UI are generally lowercase, so I think the chance for confusion about the case of the value is pretty low. As such, I am not sure it is important to make setting a `StringEnum` from value case invariant.

What do you all think?

Also, welcome back @kne42 :wave:!

**Comment by jni** [comment]
Personally, I think case invariance is a very nice feature — something that string-based mode APIs generally provide in Python. So I'd prefer to find a way to fix this. I haven't looked at the engineering properly but my gut feeling is that it's fixable.

**Comment by tlambert03** [comment]
Haven’t had time to look, but I feel like __getitem__ might be the ticket, as @kevinyamauchi suggested

**Comment by kevinyamauchi** [comment]
Okay, sounds good. I will take a look into making setting the `StringEnum` case invariant with proper error handling today and let you all know what I find! Also, it sounds like I should probably add a test or two to cover the issue that @tlambert03 found.

**Comment by kevinyamauchi** [comment]
Hello everyone. I think I have come up with something that both allows for case-invariant settings by item name or value and raises the appropriate errors when setting with invalid names/values. I have made a WIP PR to discuss at #757 (see [the tests](https://github.com/napari/napari/blob/8e5652ec47d8c063f19fc2c2271525f98ca16104/napari/util/tests/test_misc.py#L334) to get a sense for how it works). Let me know what you think!

**Comment by sofroniewn** [comment]
This has been fixed by #757
